Tribler/Core/Modules/restapi/channels/channels_torrents_endpoint.py

```diff
@@ -0,0 +1,184 @@
+import base64
+import json
+from twisted.web import http
+
+from Tribler.Core.Modules.restapi.channels.base_channels_endpoint import BaseChannelsEndpoint
+from Tribler.Core.Modules.restapi.util import convert_db_torrent_to_json
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.Core.exceptions import DuplicateTorrentFileError
+
+
+class ChannelsTorrentsEndpoint(BaseChannelsEndpoint):
+    """
+    This class is responsible for managing requests regarding torrents in a channel.
+    """
+
+    def __init__(self, session, cid):
+        BaseChannelsEndpoint.__init__(self, session)
+        self.cid = cid
+
+    def getChild(self, path, request):
+        return ChannelModifyTorrentEndpoint(self.session, self.cid, path)
+
+    def render_GET(self, request):
+        """
+        .. http:get:: /channels/discovered/(string: channelid)/torrents
+
+        A GET request to this endpoint returns all discovered torrents in a specific channel. The size of the torrent is
+        in number of bytes. The last_tracker_check value will be 0 if we did not check the tracker state of the torrent
+        yet.
+
+            **Example request**:
+
+            .. sourcecode:: none
+
+                curl -X GET http://localhost:8085/channels/discovered/da69aaad39ccf468aba2ab9177d5f8d8160135e6/torrents
+
+            **Example response**:
+
+            .. sourcecode:: javascript
+
+                {
+                    "torrents": [{
+                        "id": 4,
+                        "infohash": "97d2d8f5d37e56cfaeaae151d55f05b077074779",
+                        "name": "Ubuntu-16.04-desktop-amd64",
+                        "size": 8592385,
+                        "category": "other",
+                        "num_seeders": 42,
+                        "num_leechers": 184,
+                        "last_tracker_check": 1463176959
+                    }, ...]
+                }
+
+            :statuscode 404: if the specified channel cannot be found.
+        """
+        channel_info = self.get_channel_from_db(self.cid)
+        if channel_info is None:
+            return ChannelsTorrentsEndpoint.return_404(request)
+
+        torrent_db_columns = ['Torrent.torrent_id', 'infohash', 'Torrent.name', 'length', 'Torrent.category',
+                              'num_seeders', 'num_leechers', 'last_tracker_check', 'ChannelTorrents.inserted']
+        results_local_torrents_channel = self.channel_db_handler\
+            .getTorrentsFromChannelId(channel_info[0], True, torrent_db_columns)
+
+        results_json = [convert_db_torrent_to_json(torrent_result) for torrent_result in results_local_torrents_channel
+                        if torrent_result[2] is not None]
```

Comment:
@whirm named tuples are a nice solution, however, this requires a major refactoring in `SQLiteCacheDBHandler.py` since almost all of the methods in there are returning unnamed tuples. I would rather do this after the wx removal.